src/compiler/checker.ts

```diff
@@ -19539,7 +19546,11 @@ namespace ts {
                         }
                     }
                     if (sourceFlags & TypeFlags.Substitution) {
-                        return isRelatedTo((source as SubstitutionType).substitute, (target as SubstitutionType).substitute, RecursionFlags.Both, /*reportErrors*/ false);
+                        if (result = isRelatedTo((source as SubstitutionType).baseType, (target as SubstitutionType).baseType, RecursionFlags.Both, /*reportErrors*/ false)) {
+                            if (result &= isRelatedTo((source as SubstitutionType).constraint, (target as SubstitutionType).constraint, RecursionFlags.Both, /*reportErrors*/ false)) {
+                                return result;

```

Comment:
This is a meaningful break in behavior, right? Previously `type FilterNever<T> = T extends never ? T : never` would have allowed `FilterNever<A>` to be assignable to `FilterNever<B>` (where A and B are both type parameters), since `T & never` is just `never`, and eliminated the need to relate the type parameter part, no?

Reply:
This is in the code path for `identityRelation`, so hardly matters. In your particular example the types wouldn't relate anyway because the type parameters aren't identical.

Reply:
That's what would technically make it a break, right? Previously this related substitutions only, so `A` sub `A & never` and `B` sub `B & never` both looked identically like `never`?

Reply:
Yeah, that's true. But they definitely shouldn't be related, so I guess this fixes a bug as well.